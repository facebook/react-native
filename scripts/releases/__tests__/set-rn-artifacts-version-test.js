/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 * @format
 * @oncall react_native
 */

const readFileMock = jest.fn();
const writeFileMock = jest.fn();

jest.mock('fs', () => ({
  ...jest.requireActual<$FlowFixMe>('fs'),
  promises: {
    ...jest.requireActual<$FlowFixMe>('fs').promises,
    readFile: readFileMock,
    writeFile: writeFileMock,
  },
}));

const {REPO_ROOT} = require('../../consts');
const {updateReactNativeArtifacts} = require('../set-rn-artifacts-version');
const path = require('path');

describe('updateReactNativeArtifacts', () => {
  beforeAll(() => {
    readFileMock.mockImplementation(filePath => {
      if (
        filePath ===
        path.join(
          REPO_ROOT,
          'packages/react-native/ReactAndroid/gradle.properties',
        )
      ) {
        return 'VERSION_NAME=1000.0.0\n';
      }
    });
  });

  afterEach(() => {
    writeFileMock.mockReset();
  });

  test('should set nightly version', async () => {
    const version = '0.81.0-nightly-29282302-abcd1234';
    await updateReactNativeArtifacts(version, 'nightly');

    for (const [filePath, contents] of writeFileMock.mock.calls) {
      // Make snapshot names resilient to platform path sep differences
      expect(formatGeneratedFile(contents)).toMatchSnapshot(
        path.relative(REPO_ROOT, filePath).split(path.sep).join('/'),
      );
    }
  });

  test('should set release version', async () => {
    const version = '0.81.0';
    await updateReactNativeArtifacts(version, 'release');

    for (const [filePath, contents] of writeFileMock.mock.calls) {
      // Make snapshot names resilient to platform path sep differences
      expect(formatGeneratedFile(contents)).toMatchSnapshot(
        path.relative(REPO_ROOT, filePath).split(path.sep).join('/'),
      );
    }
  });
});

function formatGeneratedFile(source: string) {
  // Strip \@\generated annotation
  return source.replace(
    new RegExp('^ \\* @' + 'generated.*', 'gm'),
    ' * << GENERATED >>',
  );
}
