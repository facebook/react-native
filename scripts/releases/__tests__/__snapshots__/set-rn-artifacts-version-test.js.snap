// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateReactNativeArtifacts should set nightly version: packages/react-native/Libraries/Core/ReactNativeVersion.js 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict
 * << GENERATED >>
 */

const version: $ReadOnly<{
  major: number,
  minor: number,
  patch: number,
  prerelease: string | null,
}> = {
  major: 0,
  minor: 81,
  patch: 0,
  prerelease: 'nightly-29282302-abcd1234',
};

module.exports = {version};
"
`;

exports[`updateReactNativeArtifacts should set nightly version: packages/react-native/React/Base/RCTVersion.m 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * << GENERATED >>
 */

#import \\"RCTVersion.h\\"

NSString* const RCTVersionMajor = @\\"major\\";
NSString* const RCTVersionMinor = @\\"minor\\";
NSString* const RCTVersionPatch = @\\"patch\\";
NSString* const RCTVersionPrerelease = @\\"prerelease\\";


NSDictionary* RCTGetReactNativeVersion(void)
{
  static NSDictionary* __rnVersion;
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^(void){
    __rnVersion = @{
                  RCTVersionMajor: @(0),
                  RCTVersionMinor: @(81),
                  RCTVersionPatch: @(0),
                  RCTVersionPrerelease: @\\"nightly-29282302-abcd1234\\",
                  };
  });
  return __rnVersion;
}
"
`;

exports[`updateReactNativeArtifacts should set nightly version: packages/react-native/ReactAndroid/gradle.properties 1`] = `
"VERSION_NAME=0.81.0-nightly-29282302-abcd1234
"
`;

exports[`updateReactNativeArtifacts should set nightly version: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/ReactNativeVersion.java 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * << GENERATED >>
 */

package com.facebook.react.modules.systeminfo;

import com.facebook.react.common.MapBuilder;

import java.util.Map;

public class ReactNativeVersion {
  public static final Map<String, Object> VERSION = MapBuilder.<String, Object>of(
      \\"major\\", 0,
      \\"minor\\", 81,
      \\"patch\\", 0,
      \\"prerelease\\", \\"nightly-29282302-abcd1234\\");
}
"
`;

exports[`updateReactNativeArtifacts should set nightly version: packages/react-native/ReactCommon/cxxreact/ReactNativeVersion.h 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * << GENERATED >>
 */

#pragma once

#include <cstdint>
#include <string_view>

namespace facebook::react {

constexpr struct {
  int32_t Major = 0;
  int32_t Minor = 81;
  int32_t Patch = 0;
  std::string_view Prerelease = \\"nightly-29282302-abcd1234\\";
} ReactNativeVersion;

} // namespace facebook::react
"
`;

exports[`updateReactNativeArtifacts should set release version: packages/react-native/Libraries/Core/ReactNativeVersion.js 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict
 * << GENERATED >>
 */

const version: $ReadOnly<{
  major: number,
  minor: number,
  patch: number,
  prerelease: string | null,
}> = {
  major: 0,
  minor: 81,
  patch: 0,
  prerelease: null,
};

module.exports = {version};
"
`;

exports[`updateReactNativeArtifacts should set release version: packages/react-native/React/Base/RCTVersion.m 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * << GENERATED >>
 */

#import \\"RCTVersion.h\\"

NSString* const RCTVersionMajor = @\\"major\\";
NSString* const RCTVersionMinor = @\\"minor\\";
NSString* const RCTVersionPatch = @\\"patch\\";
NSString* const RCTVersionPrerelease = @\\"prerelease\\";


NSDictionary* RCTGetReactNativeVersion(void)
{
  static NSDictionary* __rnVersion;
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^(void){
    __rnVersion = @{
                  RCTVersionMajor: @(0),
                  RCTVersionMinor: @(81),
                  RCTVersionPatch: @(0),
                  RCTVersionPrerelease: [NSNull null],
                  };
  });
  return __rnVersion;
}
"
`;

exports[`updateReactNativeArtifacts should set release version: packages/react-native/ReactAndroid/gradle.properties 1`] = `
"VERSION_NAME=0.81.0
"
`;

exports[`updateReactNativeArtifacts should set release version: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/ReactNativeVersion.java 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * << GENERATED >>
 */

package com.facebook.react.modules.systeminfo;

import com.facebook.react.common.MapBuilder;

import java.util.Map;

public class ReactNativeVersion {
  public static final Map<String, Object> VERSION = MapBuilder.<String, Object>of(
      \\"major\\", 0,
      \\"minor\\", 81,
      \\"patch\\", 0,
      \\"prerelease\\", null);
}
"
`;

exports[`updateReactNativeArtifacts should set release version: packages/react-native/ReactCommon/cxxreact/ReactNativeVersion.h 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * << GENERATED >>
 */

#pragma once

#include <cstdint>
#include <string_view>

namespace facebook::react {

constexpr struct {
  int32_t Major = 0;
  int32_t Minor = 81;
  int32_t Patch = 0;
  std::string_view Prerelease = \\"\\";
} ReactNativeVersion;

} // namespace facebook::react
"
`;
